src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class SourcelinkTests : SmokeTests
+{
+    private static string SourcelinkRoot { get; } = Path.Combine(Directory.GetCurrentDirectory(), "sourcelink");
+
+    public SourcelinkTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+    /// <summary>
+    /// Verifies that all symbols have valid sourcelinks.
+    /// </summary>
+    [SkippableFact(Config.ExcludeSourcelinkEnv, skipOnTrueEnv: true)]
+    public void VerifySourcelinks()
+    {
+        if (Directory.Exists(SourcelinkRoot))
+        {
+            Directory.Delete(SourcelinkRoot, true);
+        }
+        Directory.CreateDirectory(SourcelinkRoot);
+
+        IList<string> failedFiles = ValidateAllFiles(ExtractPackages(GetAllSymbolsPackages()), GetSourcelinkToolPath());
+
+        foreach (string file in failedFiles)
+        {
+            OutputHelper.WriteLine($"Failed sourcelink verification: {file}");
+        }
+
+        Assert.True(failedFiles.Count == 0);
+    }
+
+    /// <summary>
+    /// Gets the path to sourcelink binary.
+    /// Extracts the dotnet-sourcelink tool package from PSB arhive.
+    /// </summary>
+    /// <returns>Path to sourcelink tool binary.</returns>
+    private string GetSourcelinkToolPath()
+    {
+        const string SourcelinkToolPackageNamePattern = "dotnet-sourcelink*nupkg";
+        const string SourcelinkToolBinaryFilenamePattern = "dotnet-sourcelink.dll";
```

Comment:
What makes this a _pattern_?  It appears to be the actual filename.

Reply:
Yeah, true - copy/paste issue. Will update - thanks!

Reply:
Fixed with https://github.com/dotnet/installer/pull/17309/commits/72aacfcd5be9659d9fca751b7728cdcffb13970f